src/digraph-transitivity.jl

```diff
@@ -0,0 +1,57 @@
+"""
+```transitiveclosure!(dg::DiGraph, selflooped = false)```
+
+Compute the transitive closure of a directed graph, using the Floyd-Warshall
+algorithm.
+
+Version of the function that modifies the original graph.
+
+Note: This is an O(V^3) algorithm.
+
+# Arguments
+* `dg`: the directed graph on which the transitive closure is computed.
+* `selflooped`: whether self loop should be added to the directed graph,
+default to `false`.
+"""
+function transitiveclosure!(dg::DiGraph, selflooped = false)
+    for k in vertices(dg)
+        for i in vertices(dg)
+            if i == k
+                continue
+            end
+            for j in vertices(dg)
+                if j == k
+                    continue
+                end
+                if (has_edge(dg, i, k) & has_edge(dg, k, j))
```

Comment:
`has_edge` outputs true/false, but you want a logical AND, not a boolean one. A boolean AND is not short-circuiting - that is, both arguments will be evaluated:

```
julia> function foo()
         println("foo")
         return true
       end
foo (generic function with 1 method)

julia> foo()
foo
true

julia> false && foo()   # short-circuits - no println.
false

julia> false & foo()  # does not short-circuit.
foo
false
```

You want to prefer short-circuiting wherever possible since you don't need to execute the second `has_edge` if you know the first one doesn't exist.